src/components/show-products/index.js

```diff
@@ -16,9 +16,33 @@ export const ShowProducts = () => {
   const { products, loading } = useProducts();
   const { push } = useHistory();
   const { storedValue } = useLocalStorage(LOCAL_STORAGE_LIST_TOKEN);
-  const timeFrames = TimeFrames;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [list, setList] = useState([]);
 
-  const handleOnChange = (productID) => {
+  useEffect(() => {
+    let results = products;
+
+    if (searchTerm !== '') {
+      results = products.filter(({ productName }) =>
```

Comment:
Nit: This could be refactored in a way that you don't need to mutate `results`.... One way to achieve this is to move this block into its separate function:

e.g.

```
function getFilteredProducts(products) {
  if (searchTerm === '') {
    return products;
  }

  return products.filter(....);
}

// Then inside this `useEffect`, you can assign directly the result of that function to `results`.
const results = getFilteredProducts(products);
setList(results);
```

By doing this, the indentation is reduced, meaning that readability is improved for other developers working in the project (and even for you in the future) 🙂.

**Disclaimer:** This is just a suggestion/idea for improvement, but no need to use this if you don't find it useful.